incubator/hnc/pkg/controllers/setup.go

```diff
@@ -13,20 +13,35 @@ import (
 // Create creates all reconcilers.
 //
 // This function is called both from main.go as well as from the integ tests.
-func Create(mgr ctrl.Manager, f *forest.Forest, maxReconciles int) error {
+func Create(mgr ctrl.Manager, f *forest.Forest, maxReconciles int, newObjectController bool) error {
 	// Create all object reconcillers
 	objReconcilers := []NamespaceSyncer{}
-	for _, gvk := range config.GVKs {
-		or := &ObjectReconciler{
-			Client: mgr.GetClient(),
-			Log:    ctrl.Log.WithName("controllers").WithName(gvk.Kind),
-			Forest: f,
-			GVK:    gvk,
+	if newObjectController {
```

Comment:
I feel like this might be easier to read if it were something like:

```
or := createObjectReconciler(newObjectController, mgr, f, gvk)
```

because it's not immediately clear from reading this that the _only_ thing that changes in these two code blocks is the type of the reconciler.